app/enterprise/1.5.x/kong-for-kubernetes/deployment-options.md

```diff
@@ -0,0 +1,179 @@
+
+In traditional deployments, Kong's database (PostgreSQL or Cassandra) provides
+a persistent store of configuration available to all Kong nodes to ensure
+consistent proxy behavior across the cluster that is not affected by node
+restarts. Because etcd provides this functionality in Kong for Kubernetes
+deployments, it is not necessary to run an additional database, reducing
+maintenance and infrastructure requirements.
+
+While Kong for Kubernetes does not require a database, it is fully compatible
+with PostgreSQL and requires it for some features. etcd remains the source of
+truth, but controller can translate Kubernetes resources there into either
+DB-less or database configuration.
+
+# Choosing between DB-less or database-backed deployments
+
+In general, DB-less deployments are simpler to maintain and require less
+resources to run, and as such are the preferred option for Kong for Kubernetes.
+These deployments use the `kong-enterprise-k8s` image and must set
+`KONG_DATABASE=off` in their environment variables.
+
+Database-backed deployments offer a wider range of features using the
+`kong-enterprise-edition` image. Review the sections below to determine if your
+use case requires a feature that is not available in DB-less deployments.
+
+## Feature availability
+
+Some Kong Enterprise features are not available in DB-less deployments. Users
+should use the `kong-enterprise-edition` image and a database-backed deployment
+if they wish to use:
+
+* Kong Manager
+* Dev Portal
+* Brain
+* Immunity
+* Teams (RBAC)
+* Vitals
+* Workspaces
+
+Because Kong for Kubernetes is configured by the ingress controller, some
+functionality in these features is different from traditional deployments:
+
+* Users will not typically create proxy configuration through Kong Manager;
+  proxy configuration is normally managed by the controller and users provide
+  configuration via Kubernetes resources.
+* Because the controller creates proxy configuration on behalf of users, users
+  do not need to interact with the Admin API directly. Typical Kong for
+  Kubernetes deployments do not expose the Admin API to protect it in lieu of
+  RBAC: only the controller can access it. Kubernetes-level RBAC rules and
+  namespaces should be used to restrict what configuration administrators can
+  create.
+* Ingress controller instances create configuration in a single workspace only
+  (`default` by default). To use multiple workspaces, users should deploy
+  multiple controller instances, setting the `CONTROLLER_KONG_WORKSPACE`
+  environment variable to the workspace that instance should use. These
+  instances should set `CONTROLLER_INGRESS_CLASS` to unique values, to avoid
+  creating duplicate configuration in workspaces. Note that if controller
+  instances are deployed outside the Kong pod the Admin API must be exposed,
+  and users should enable RBAC with workspace admin users for the controllers.
+  Set `CONTROLLER_KONG_ADMIN_TOKEN` to the RBAC user's token.
```

Comment:
This entire bullet on how to set up instances as workspaces should really be a separate section with steps, though I'm not sure that a section like that would belong in this topic. Maybe shorten this bullet to something like:

Ingress controller instances create configuration in a single workspace only (`default` by default). To use multiple workspaces, deploy multiple controller instances. See [the section on doing this thing] for more detail.

Reply:
+1 for dedicated sections

Reply:
We don't currently have a good way of doing this or instructions for it--the only option is to cut out the controller section of one of our stock manifests, wrap it in its own deployment, and deploy it independently, which we don't really want users to do (it's tedious and leaves a lot of room for error).

We do plan to [add a one-step setting in the chart](https://trello.com/c/ClRh4pvm/389-support-controller-only-releases) to disable the non-controller components of the deployment, and should probably create a static manifest for a controller-only deployment also. Would it make sense to leave this as-is for now, write the new doc along with those manifest changes, and link it once done?